src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.csproj

```diff
@@ -10,7 +10,9 @@
   <PropertyGroup>
     <IsPackable>true</IsPackable>
     <VersionPrefix>4.3.0</VersionPrefix>
-    <VersionPrefix Condition="'$(IsPackable)' == 'true'">4.4.0</VersionPrefix>
+    <VersionPrefix Condition="'$(IsPackable)' == 'true'">4.7.0</VersionPrefix><!-- Ensures AssemblyInformationalVersion is larger than the last released -->
+    <AssemblyVersion>4.0.2.0</AssemblyVersion>
+    <AssemblyVersion Condition="'$(IsPackable)' == 'true'">4.0.3.0</AssemblyVersion>
```

Comment:
Why are you setting AssemblyVersion explicitly now? I don't think that should be necessary.

Reply:
I'm seeing the following:

Before this change (nuget):
- [assembly: AssemblyFileVersion("4.6.24705.01")]
- [assembly: AssemblyInformationalVersion("4.6.24705.01. Commit Hash: 4d1af962ca0fede10beb01d197367c2f90e92c97")]
- [assembly: AssemblyVersion("4.0.2.0")]

Without this change (main):
- [assembly: AssemblyFileVersion("4.400.24.52904")]
- [assembly: AssemblyInformationalVersion("4.4.0-preview.1.24529.4+c217b3e67dea4a77c604440fdf5ee9197565fbee")]
- [assembly: AssemblyVersion("4.4.0.0")]

With this change (PR):
- [assembly: AssemblyFileVersion("4.700.24.52901")]
- [assembly: AssemblyInformationalVersion("4.7.0-dev.24529.1+f2633cbdb1640f2588165f27e9a97a113ec3780f")]
- [assembly: AssemblyVersion("4.0.3.0")]

Reasoning:
- I don't think I should be manually setting AssemblyFileVersion. Seems autogenerated.
- AssemblyInformationalVersion seems to use VersionPrefix. That number was wrong before (went from 4.6 to 4.4, now it's 4.7). So that bug is fixed.
- AssemblyVersion as you can see was changing fine before (4.0.2.0 to 4.4.0.0), but it was not increasing consistently with the rest of the packages in the repo. So I set it manually so the increase was on the Minor version, not in the Major version. Do you think I should not do this?

Reply:
Hashcode's assembly version also gets bumped across major versions: 6.0.0.0. https://github.com/dotnet/maintenance-packages/blob/e9d1a907484bfb3b98e0b66949a628c8d768d1e1/src/Microsoft.Bcl.HashCode/src/Microsoft.Bcl.HashCode.csproj#L14

Reply:
That's the VersionPrefix, not the AssemblyVersion.

Reply:
I think the AssemblyVersion doesn't need to be modified anyway, since it was increased correctly, even if it skipped numbers. I'll remove it.

Reply:
If I remove the AssemblyVersion lines, now the value that gets automatically selected is:

- [assembly: AssemblyVersion("4.7.0.0")]

Reply:
> That's the VersionPrefix, not the AssemblyVersion.

The AssemblyVersion is calculated based on the VersionPrefix: https://github.com/dotnet/arcade/blob/e4df2888e714255352a6afa2ec1246e70ac21a49/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.targets#L23

So when I linked to it, I meant that for Hashcode the assembly version is also increased in a "major" way and that's fine.

Reply:
> [assembly: AssemblyVersion("4.7.0.0")]

Correct, that's fine.